robot/pmm/resources/pmm.py

```diff
@@ -274,3 +316,83 @@ def verify_toast_message(self, message):
+        """During winter 2020 part of the modal fields appear as lightning elements.
+        This keyword validates , identifies the element and populates value"""
+        for key, value in kwargs.items():
+            if key in ("Start Date", "End Date", "Delivery Date"):
+                locator = pmm_lex_locators["new_record"]["c_lightning_datepicker"].format(key)
+                if self.check_if_element_exists(locator):
+                    element = self.selenium.driver.find_element_by_xpath(locator)
+                    self.selenium.driver.execute_script("arguments[0].scrollIntoView(true)",element)
+                   # self.selenium.scroll_element_into_view(locator)
+                    self.selenium.wait_until_element_is_visible(locator)
+                    self.selenium.set_focus_to_element(locator)
+                    self.open_date_picker(key)
+                    self.pick_date(value)
+                else:
+                    self.builtin.log(f"Element {key} not found")
+
+            elif key in ("Status", "Program Issue Area", "Stage", "Role"):
+                locator = pmm_lex_locators["new_record"]["c_dd_popup"].format(key)
+                selection_value = pmm_lex_locators["new_record"]["c_dd_selection"].format(value)
+                if self.check_if_element_exists(locator):
+                    self.selenium.set_focus_to_element(locator)
+                    self.selenium.wait_until_element_is_visible(locator)
+                    self.selenium.scroll_element_into_view(locator)
+                    self.salesforce._jsclick(locator)
+                    self.selenium.wait_until_element_is_visible(selection_value)
+                    self.selenium.click_element(selection_value)
+
+            elif key in ("Program", "Client", "Program Engagement", "Service", "Household Account", "Program Cohort"):
+                self.salesforce.populate_lookup_field(key, value)
+
+            elif key in ("Service Name", "Quantity", "Description", "Unit of Measurement", "Target Population", "Program Cohort"):
+                locator = pmm_lex_locators["new_record"]["c_field-input"].format(key)
+                self.salesforce._populate_field(locator, value)
+
+            elif key in ("Auto-Name Override", "Auto-name Override"):
+                locator = pmm_lex_locators["new_record"]["override_checkbox"].format(key)
+                if value == "checked":
+                    self.selenium.get_webelement(locator).click()
+            else:
+                raise Exception("Locator not found")
+
+
+    def open_date_picker(self, title):
+        """Opens the date picker by clicking on the date picker icon given the title of the field"""
+        if self.latest_api_version == 50.0:
+            locator = pmm_lex_locators["new_record"]["c_lightning_datepicker"].format(
+                title
+            )
+        else:
+            locator = pmm_lex_locators["new_record"]["c_open_date_picker"].format(
```

Comment:
Is this locator being used somewhere else? if not then just move this to locator 49 file with same name so that u can avoid this if else logic

Reply:
Removed the if logic